lib/validator/common.go

```diff
@@ -80,3 +82,39 @@ func ValidateMetadataIDsAssigned(rm unversioned.ResourceMetadata) error {
 
 	return nil
 }
+
+// ValidateIDFieldsSame is used to validate that the Resource Objects
+// supplied in one slice reference the same objects in the other slice
+// by comparing the ID fields.
+func ValidateIDFieldsSame(original, changed []unversioned.ResourceObject) error {
+	toCompare := make(map[string]int)
+	for i, obj := range original {
+		toCompare[obj.GetResourceID()] = i
+	}
+
+	for _, obj := range changed {
+		if _, ok := toCompare[obj.GetResourceID()]; !ok {
+			return fmt.Errorf("Cannot change metadata fields: %v, resource will not be found", GetIDFields(obj.GetResourceMetadata()))
+		}
+	}
+
+	return nil
+}
+
+// GetIDFields is a helper function that returns the field labels
+// on the metadata objects that are ID fields excluding Labels,
+// Annotations, and Tags
+func GetIDFields(rm unversioned.ResourceMetadata) []string {
```

Comment:
I don't like this.  I think if you really want to get a slice of IDs as strings, then the getResourceID() should probably have been getResourceIDs() returning a slice of strings. 

*Personally* I don't think that's necessary - if you have a method that returns a nice user friendly name of the resource then you can craft some text that says  Resource X should be Resource Y.